### PR TITLE
Remove nixops and deployment server

### PR DESCRIPTION
**Pull request by shmish111**
The deployment server can be replaced by a github ssh action.
After this PR I plan to simplify the terraform code but this is the first step.

Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: deployment/morph/prometheus.nix [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+            ];
+          }
+          {
+            job_name = "prometheus";
+            scrape_interval = "10s";
+            static_configs = [
+              {
+                targets = [
+                  "localhost:9090"
```
Comment (gilligan) [inline deployment/morph/prometheus.nix:89]:
```suggestion
                  "localhost:${prometheusPort}"
```

File: deployment/morph/webghc.nix [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
       networking.firewall = {
         enable = true;
-        allowedTCPPorts = [ 80 9100 9091 9113 ];
+        allowedTCPPorts = [ 22 80 9100 9091 9113 ];
```
Comment (gilligan) [inline deployment/morph/webghc.nix:36]: I'm seeing `allowedTCPPorts` in several places, i wonder if it would be feasible to extract/import it? It's probably fine either way speaking of myself i certainly remember banging my head against the wall because of some screw-up with conflicting port specifications..

Just a thought. Maybe anything "smarter" than you have now just makes it more complicated. I leave that up to you

File: deployment/morph/webghc.nix [hunk 3]
```diff
@@ -52,15 +51,15 @@
...
+        openssh.authorizedKeys.keys = monitoringKeys;
         packages = [ serviceSystemctl ];
       };
 
       systemd.services.web-ghc = {
         wantedBy = [ ];
         before = [ ];
         enable = true;
-        path = [ "${web-ghc}" ];
+        path = [ ];
```
Comment (gilligan) [inline deployment/morph/webghc.nix:62]: You can omit the empty path because `[ ]` is the default value anyhow. Same goes for `wantedBy` and `before`.
What you could maybe *add* though is

```
after = [ "network.target" ];
```

File: plutus-ledger/plutus-ledger.cabal [hunk 4]
```diff
@@ -80,7 +80,7 @@ library
     build-depends:
         base >=4.9 && <5,
         aeson -any,
-        base16-bytestring -any,
+        base16-bytestring <1,
```
Comment (gilligan) [inline plutus-ledger/plutus-ledger.cabal:83]: Is this related to the rest of this PR?

**Review by gilligan — CHANGES_REQUESTED** [review]

File: plutus-ledger/plutus-ledger.cabal (hunk 4 above)
Comment (shmish111) [reply plutus-ledger/plutus-ledger.cabal:83]: only in that it is required when removing the deployment server as that haskell component was forcing this restriction to be the case.

File: deployment/morph/webghc.nix (hunk 2 above)
Comment (shmish111) [reply deployment/morph/webghc.nix]: ok I've added a `ports` object to configuration.nix which I now thread through. This should avoid any problems using incorrect ports in the future.

**Review by gilligan — APPROVED** [review]
LGTM! nice job! :)
